node/node-core/src/eth/mod.rs

```diff
@@ -0,0 +1,263 @@
+use crate::{
+    error::RpcError,
+    eth::{call::CallRequest, filter::Filter, transaction::EthTransactionRequest},
+    types::Index,
+};
+use ethers_core::{
+    abi::ethereum_types::H64,
+    types::{Address, BlockNumber, Bytes, Transaction, TxHash, H256, U256},
+};
+use serde::{de::DeserializeOwned, Deserialize, Deserializer, Serialize};
+
+pub mod block;
+pub mod call;
+pub mod filter;
+pub mod receipt;
+pub mod transaction;
+pub mod trie;
+pub mod utils;
+
+/// Represents ethereum JSON-RPC API
+#[derive(Clone, Debug, PartialEq, Deserialize)]
+#[serde(tag = "method", content = "params")]
+pub enum EthRequest {
```

Comment:
beautiful - this is the stuff which maybe makes sense to be in a separate "eth json rpc axum" crate